app/components/post_list/post/body/content/permalink_preview/index.tsx

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {withDatabase, withObservables} from '@nozbe/watermelondb/react';
+import {of as of$} from 'rxjs';
+import {switchMap, distinctUntilChanged} from 'rxjs/operators';
+
+import {fetchUsersByIds} from '@actions/remote/user';
+import {withServerUrl} from '@context/server';
+import {observePost} from '@queries/servers/post';
+import {observeConfigBooleanValue} from '@queries/servers/system';
+import {observeUserOrFetch, observeTeammateNameDisplay, observeCurrentUser} from '@queries/servers/user';
+
+import PermalinkPreview from './permalink_preview';
+
+import type {WithDatabaseArgs} from '@typings/database/database';
+
+const enhance = withObservables(['embedData', 'serverUrl'], ({database, embedData, serverUrl}: WithDatabaseArgs & {embedData: PermalinkEmbedData; serverUrl?: string}) => {
+    const showPermalinkPreviews = observeConfigBooleanValue(database, 'EnablePermalinkPreviews', false);
```

Comment:
most of the things here should not be processed if the value for this is false, don't you think ?

if that is the case, perhaps this entire component should not even be mounted